Makefile.btfgen

```diff
@@ -0,0 +1,45 @@
+# disable builtin rules (avoid trying to run cc for .o files)
+MAKEFLAGS += --no-builtin-rules
+
+# SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause)
+SOURCE_BTF_FILES = $(shell find $(BTFHUB_ARCHIVE)/ -iregex ".*$(subst x86,x86_64,$(ARCH)).*" -type f -name '*.btf.tar.xz')
+MIN_CORE_BTF_FILES = $(patsubst $(BTFHUB_ARCHIVE)/%.btf.tar.xz, $(OUTPUT)/%.btf, $(SOURCE_BTF_FILES))
+# TODO (filter by architecture too when implemented)
+BPF_O_FILES = $(shell find pkg/gadgets -type f -name '*.o')
+
+ifndef BTFHUB_ARCHIVE
+$(error BTFHUB_ARCHIVE is undefined)
+endif
+
+ifndef ARCH
+$(error ARCH is undefined)
+endif
```

Comment:
Shouldn't it be at the top to be checked before the `find` command is executed? It seems to work like this but I don't understand why.